server/src/main/java/org/elasticsearch/search/aggregations/support/SamplingContext.java

```diff
@@ -73,4 +82,22 @@ public double inverseScale(double value) {
         return value;
     }
 
+    public Query buildQueryWithSampler(QueryBuilder builder, AggregationContext context) throws IOException {
```

Comment:
Javadoc on both of these please.  They have similar names but do different things.